lib/components/viewers/live-stop-times.tsx

```diff
@@ -151,35 +159,143 @@ class LiveStopTimes extends Component<Props, State> {
+      .filter(({ pattern, route }) =>
+        routeIsValid(route, getRouteIdForPattern(pattern))
+      )
+
+    // Determine if arrival occurs on different day, making sure to account for
+    // any extra days added to the service day if it arrives after midnight. Note:
+    // this can handle the rare (and non-existent?) case where an arrival occurs
+    // 48:00 hours (or more) from the start of the service day.
+
+    const now = utcToZonedTime(Date.now(), homeTimezone)
+    const tomorrow = addDays(now, 1)
+    const dayAfterTomorrow = addDays(now, 2)
+
+    const findTimes = (times: any, date: any): any[] => {
+      return times
+        .map((route: any) => {
+          return {
+            ...route,
+            times: route.times.filter((time: Time) => {
+              const departureTime = time.realtimeDeparture
+              const serviceDay = utcToZonedTime(
+                new Date(time.serviceDay * 1000),
+                homeTimezone
+              )
+              const departureTimeRemainder = departureTime % ONE_DAY_IN_SECONDS
+              const daysAfterServiceDay =
+                (departureTime - departureTimeRemainder) / ONE_DAY_IN_SECONDS
+              const departureDay = addDays(serviceDay, daysAfterServiceDay)
+              return isSameDay(date, departureDay)
+            })
+          }
+        })
+        .filter((route: any) => route.times.length > 0)
+    }
+
+    const todayTimes = findTimes(routeTimes, now)
+    const tomorrowTimes = findTimes(routeTimes, tomorrow)
+    const dayAfterTomorrowTimes = findTimes(routeTimes, dayAfterTomorrow)
+
     return (
       <>
-        <ul className="route-row-container">
-          {Object.values(stopTimesByPattern)
-            .sort(patternComparator)
-            .filter(({ pattern, route }) =>
-              routeIsValid(route, getRouteIdForPattern(pattern))
-            )
-            .map(({ id, pattern, route, times }) => {
-              // Only add pattern if route info is returned by OTP.
-              return (
-                <PatternRow
-                  homeTimezone={homeTimezone}
-                  key={id}
-                  pattern={pattern}
-                  route={{
-                    ...route,
-                    operator: transitOperators.find(
-                      (o: TransitOperator) => o.agencyId === route.agencyId
-                    )
-                  }}
-                  showOperatorLogo={showOperatorLogo}
-                  stopTimes={times}
-                  stopViewerArriving={stopViewerArriving}
-                  stopViewerConfig={stopViewerConfig}
+        <div>
+          {todayTimes.length > 0 && (
+            <div className="list-container">
+              <ul className="route-row-container">
+                {todayTimes.map(({ id, pattern, route, times }) => {
+                  // Only add pattern if route info is returned by OTP.
+                  return (
+                    <PatternRow
```

Comment:
I don't like repeating the same code 3 times. Is there any way to replace the 3 arrays with a new field inside of a new single array that's date or something, we can then check for that date changing inside of the map and render a heading accordingly. Let me know if you'd like to discuss this more offline, I'm not sure if that was super clear